File: src/main/java/com/zunza/buythedip/user/repository/UserRepository.java
package com.zunza.buythedip.user.repository;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import com.zunza.buythedip.user.entity.User;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {
	boolean existsByAccountId(String accountId);
	boolean existsByNickname(String nickname);
	Optional<User> findByAccountId(String accountId);
}

File: src/main/kotlin/com/zunza/buythedip/user/repository/UserRepository.kt
package com.zunza.buythedip.user.repository

import com.zunza.buythedip.user.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : JpaRepository<User, Long> {
    fun findByEmail(email: String): User?
    fun existsByEmail(email: String): Boolean
    fun existsByNickname(nickname: String): Boolean
    fun findByProviderId(providerId: Long): User?
}
